modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectEntryLocalServiceImpl.java

```diff
@@ -811,6 +813,53 @@ public int getObjectEntriesCount(long groupId, long objectDefinitionId) {
 		return objectEntryPersistence.countByG_ODI(groupId, objectDefinitionId);
 	}
 
+	@Transactional(propagation = Propagation.REQUIRES_NEW)
+	public long getObjectEntriesCount(
+			long groupId, ObjectDefinition objectDefinition,
+			Predicate predicate)
+		throws PortalException {
+
+		DynamicObjectDefinitionTable dynamicObjectDefinitionTable =
+			_getDynamicObjectDefinitionTable(
+				objectDefinition.getObjectDefinitionId());
+
+		DynamicObjectDefinitionTable extensionDynamicObjectDefinitionTable =
+			_getExtensionDynamicObjectDefinitionTable(
+				objectDefinition.getObjectDefinitionId());
+
+		JoinStep joinStep = DSLQueryFactoryUtil.countDistinct(
+			dynamicObjectDefinitionTable.getPrimaryKeyColumn()
+		).from(
+			dynamicObjectDefinitionTable
+		).innerJoinON(
+			extensionDynamicObjectDefinitionTable,
+			extensionDynamicObjectDefinitionTable.getPrimaryKeyColumn(
+			).eq(
+				dynamicObjectDefinitionTable.getPrimaryKeyColumn()
+			)
+		);
+
+		if (!StringUtil.equals(objectDefinition.getScope(), "site")) {
+			return dslQueryCount(joinStep.where(predicate));
+		}
+
+		return dslQueryCount(
+			joinStep.innerJoinON(
+				ObjectEntryTable.INSTANCE,
+				ObjectEntryTable.INSTANCE.objectEntryId.eq(
+					dynamicObjectDefinitionTable.getPrimaryKeyColumn())
+			).where(
+				predicate.and(
+					() -> {
+						if (groupId == 0) {
```

Comment:
It seems like this part of the query just will be appended if the scope demands a groupId != 0. I believe this verification is redundant.

Reply:
Aewsome catch!
Thanks for the tip!